File: api/lib/units/artillery.rb
require_relative './vehicle.rb'

class Artillery < Vehicle
  def initialize square, team
    super
    @type = :artillery
  end

  def assign_threat
    super
    @square
      .neighbors(headings: [[0, 2], [0, 3], [1, 2], [2, 1]])
      .each { |square| threaten square }
  end
end

File: api/src/classes/units/Artillery.js
const Vehicle = require('./vehicle')

class Artillery extends Vehicle {
  constructor(square, team) {
    super(square, team)
    this.type = 'artillery'
  }

  assignThreat() {
    super()
    this.square
      .neighbors({headings: [[0, 2], [0, 3], [1, 2], [2, 1]]})
      .forEach(square => threaten(square))
  }
}

module.exports = Artillery